src/main/java/codoc/model/person/Year.java

```diff
@@ -0,0 +1,59 @@
+package codoc.model.person;
+
+import static codoc.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Person's Year in CoDoc.
+ * Guarantees: immutable; is valid as declared in {@link #isValidYear(String)}
+ */
+public class Year {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Years should only contain number from 1-4, and it should not be blank";
+
+    /*
+     * The first character of the Year must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "^[1-4]$";
+
+    public final String year;
```

Comment:
Year represented as String and regex limits it to be entered as an integer between 1~4.